### PR TITLE
AVL Tree implemented in Dart

### LINKED ISSUE #2112: AVL tree
- [x] C
- [x] C++ - @Manish-cloud
- [ ] C# - @KanakalathaVemuru
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go
- [x] Java
- [ ] JS - @satvikchachra
- [ ] Kotlin
- [ ] Php - @chandrikadeb7
- [x] Python
- [x] Readme
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Rukmini-Meda**
Fixes #2112 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- AVL Tree is implemented in Dart.
- Insertion and deletion operations are implemented.

#### Languages Used:
- Dart

#### Files Added:
- AVL_Tree.dart

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: AVL_Tree/AVL_Tree.dart [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+ * 
+ * AVL tree is preferred when insertions and deletions are less frequent and search is a
+ * more frequent operation.
+ * 
+ * 
+ */
+
+// Importing required libraries
+import 'dart:io';
+
```
Comment (singh-shreya6) [inline AVL_Tree/AVL_Tree.dart:19]: These is no need of extra line here.

File: AVL_Tree/AVL_Tree.dart [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+import 'dart:math';
+
+// Class definition for a tree node.
+class Node{
+
+    int value;
+
```
Comment (singh-shreya6) [inline AVL_Tree/AVL_Tree.dart:26]: Please remove extra blank line.

File: AVL_Tree/AVL_Tree.dart [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+    Node left , right;
+
```
Comment (singh-shreya6) [inline AVL_Tree/AVL_Tree.dart:28]: Please remove extra blank line.

File: AVL_Tree/AVL_Tree.dart [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+        Node c2 = root.left.right;
+
+        y.right = z;
+
+        y.left = x;
+
+        z.left = c2;
+
+        y.height = 1 + max<int>(height(y.left),height(y.right));
+
```
Comment (singh-shreya6) [inline AVL_Tree/AVL_Tree.dart:105]: please give space after comma

File: AVL_Tree/AVL_Tree.dart [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+        Node c2 = y.left;
+
+        y.left = z;
+
+        y.right = x;
+
+        z.right = c2;
+
+        y.height = 1 + max<int>(height(y.left),height(y.right));
```
Comment (singh-shreya6) [inline AVL_Tree/AVL_Tree.dart:127]: please give space after comma

File: AVL_Tree/AVL_Tree.dart [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+ * 3
+ * 30
+ * 20
+ * 10
+ * 25
+ * 40
+ * Enter 1 to continue:
+ * 0
+ * 
+ */
```
Comment (singh-shreya6) [inline AVL_Tree/AVL_Tree.dart:473]: please add blank line at end of file

**Comment by Rukmini-Meda** [comment]
@singh-shreya6 I have made changes as suggested by you. Please merge.

**Comment by Rukmini-Meda** [comment]
@singh-shreya6 I have squashed the commits. Please merge. Also, the issue this PR fixes is #2112 and it is of medium difficulty. Please add the medium difficulty tag.

**Pull request merged**